src/chunks/vocab/subword.rs

```diff
@@ -409,12 +410,63 @@ impl SubwordVocab<ExplicitIndexer> {
             .map_err(|e| ErrorKind::io_error("Cannot write maximum n-gram length", e))?;
 
         write_vocab_items(write, self.words())?;
-        write_vocab_items(write, self.indexer.ngrams())?;
+        let indexed_ngrams_iter = self.indexer.ngrams().iter().map(|ngram| {
+            (
+                ngram.as_str(),
+                self.indexer
+                    .index_ngram(&ngram.as_str().into())
+                    .expect("Indexers need to index all their ngrams."),
```

Comment:
Agreed, changed the wording.

I moved whole `iter.().map()` thing to the write method, looks much cleaner. I replaced the `expect` with a `Result` since we're returning results all around anyways.